assignments/tests/test_views.py

```diff
@@ -60,27 +65,101 @@ def test_all_budget_tasks_present_in_section_for_open_assignment(self, create_as
                     break
             assert found, 'Cannot find budgeting task {} in the response'.format(task.name)
 
-        @pytest.mark.django_db
-        def test_all_budget_targets_present_in_budget_tasks_for_open_assignment(self, create_assignments,
-                                                                                assignments_url):
-            assignment = Assignment.objects.filter(status=Assignment.STATUS_OPEN).first()
-            sections = Section.objects.filter(assignment__id=assignment.id)
-            for section in sections:
-                api_client = APIClient()
-                section_url = reverse('assignments:section-detail', args=[assignment.slug, section.id])
-                response = api_client.get(section_url)
-                response_data = response.json()
+    @pytest.mark.django_db
+    def test_all_budget_targets_present_in_budget_tasks_for_open_assignment(self, create_assignments,
+                                                                            assignments_url):
+        assignment = Assignment.objects.filter(status=Assignment.STATUS_OPEN).first()
+        sections = Section.objects.filter(assignment__id=assignment.id)
+        for section in sections:
+            api_client = APIClient()
+            section_url = reverse('assignments:section-detail', args=[assignment.slug, section.id])
+            response = api_client.get(section_url)
+            response_data = response.json()
+
+            for budget_target in BudgetingTarget.objects.filter(budgeting_tasks__section=section):
+                found_target = False
+                for budget_task in budget_target.budgeting_tasks.all():
+                    for data_budget_task in response_data['budgeting_tasks']:
+                        if budget_task.name == data_budget_task['name']:
+                            for data_target in data_budget_task['targets']:
+                                if data_target['name'] == budget_target.name:
+                                    found_target = True
+                                    break
```

Comment:
Sorry for being that negative. Just tried to think if there were some simpler way of doing this since it is pretty hard to read at the moment.

Could you something like this:
```
# Get the targets you want to check are present in the response
targets_saved_in_db_ids = BudgetingTarget.objects.filter(...).values_list('id')
# Get the targets in the JSON response (not sure about the structure so poor example)
targets_in_response_ids = ...
# assert the two lists have equal items. Use for example: https://docs.python.org/3.2/library/unittest.html#unittest.TestCase.assertCountEqual
assertCountEqual(targets_saved_in_db_ids, targets_in_response_ids)
```